### PR TITLE
Don't allow conflicting float tolerances

### PR DESCRIPTION
**Pull request by evouga**

### CODE REVIEW

File: spec/2023-07-draft.md [hunk 1]
```diff
@@ -662,11 +662,12 @@ It supports the following command-line arguments to control how tokens are compa
 | `float_absolute_tolerance ε` | indicates that floating-point tokens should be accepted if they are within absolute error ≤ ε (see below for details).                                      |
 | `float_tolerance ε`          | short-hand for applying ε as both relative and absolute tolerance.                                                                                          |
 
-It is an error to specify `float_tolerance` together with `float_relative_tolerance` and/or `float_absolute_tolerance`.
-
 When supplying both a relative and an absolute tolerance, the semantics are that a token is accepted if it is within either of the two tolerances.
 When a floating-point tolerance has been set, any valid formatting of floating point numbers is accepted for floating point tokens.
 So, for instance, if a token in the answer file says `0.0314`, a token of `3.14000000e-2` in the output file would be accepted.
+
+It is an error to provide any of the `float_relative_tolerance`, `float_absolute_tolerance`, or `float_tolerance` arguments more than once, or to provide a `float_tolerance` alongside `float_relative_tolerance` and/or `float_absolute_tolerance`.
```
[the remark this answers is not in the document]
Comment (evouga) [reply spec/2023-07-draft.md:669]: Isn't `output_validator_flags` allowed to be an arbitrary string?

[the remark this answers is not in the document]
Comment (RagnarGrootKoerkamp) [reply spec/2023-07-draft.md:669]: Ah my bad; yes you are right.

**Comment by Tagl** [comment]
With the new text, the implementation of the default validator used by Kattis, and probably other systems too, will require changes.
https://github.com/Kattis/problemtools/blob/e8640fd9c617dfa5348ec331fc19d78b02f20904/support/default_validator/default_validator.cc#L85-L107
Currently it overwrites and processes arguments in order, first to last.
Good to specify the behaviour though.

**Comment by niemela** [comment]
> With the new text, the implementation of the default validator used by Kattis, and probably other systems too, will require changes.

If that's what current systems do, I think we should specify it that way.

It doesn't really matter which way it's specified... but it should be specified, and in these cases I would lean towards "whatever the implementers have done"

**Comment by evouga** [comment]
I am little less ambivalent, since:
- I don't see a realistic scenario where multiple copies of a tolerance argument are intended/useful;
- I *do* see a realistic scenario where it causes problems during a contest, i.e.: a problem package accidentally defines a second copy of `float_relative_tolerance` instead of `float_absolute_tolerance`, resulting in both tolerances having unexpected values.

**Review by pehrsoderman — APPROVED** [review]
I think it's better to be strict here, and and error out on an obviously incorrect configuration.

**Pull request merged**